pymc_marketing/mmm/base.py

```diff
@@ -1293,6 +1295,130 @@ def plot_channel_contribution_share_hdi(
     def graphviz(self, **kwargs):
         return pm.model_to_graphviz(self.model, **kwargs)
 
+    def _process_decomposition_components(self, data: pd.DataFrame):
+        """
+        Process data to compute the sum of contributions by component and calculate their percentages.
+        Parameters
+        ----------
+        dataframe : pd.DataFrame
+            Dataframe containing the time series data.
+        Returns
+        -------
+        dataframe : pd.DataFrame
+            A dataframe with contributions summed up by component, sorted, and with percentages.
+        """
+        dataframe = data.copy()
+        stack_dataframe = dataframe.stack().reset_index()
+        stack_dataframe.columns = pd.Index(["date", "component", "contribution"])
+        stack_dataframe.set_index(["date", "component"], inplace=True)
+        dataframe = stack_dataframe.groupby("component").sum()
+        dataframe.sort_values(by="contribution", ascending=True, inplace=True)
+        dataframe.reset_index(inplace=True)
+
+        total_contribution = dataframe["contribution"].sum()
+        dataframe["percentage"] = (dataframe["contribution"] / total_contribution) * 100
+
+        return dataframe
+
+    def plot_waterfall_components_decomposition(
+        self,
+        original_scale: bool = True,
+        figsize: Tuple = (14, 7),
```

Comment:
can you use `Tuple[int, int]` to be more explicit :)

Reply:
Actually, after https://github.com/pymc-labs/pymc-marketing/pull/635  we should use `tuple`

Reply:
Change!